src/js/view/controls/controls.ts

```diff
@@ -25,24 +28,158 @@ const ACTIVE_TIMEOUT = OS.mobile ? 4000 : 2000;
+        this.inactiveTime = 0;
+    }
+
+    userActive(timeout: number = ACTIVE_TIMEOUT): void {
+        if (timeout > 0) {
+            this.inactiveTime = now() + timeout;
+            if (this.activeTimeout === -1) {
+                this.activeTimeout = setTimeout(this.userInactiveTimeout, timeout);
+            }
+        } else {
+            this.resetActiveTimeout();
+        }
+        if (!this.showing) {
+            removeClass(this.playerContainer, 'jw-flag-user-inactive');
+            this.showing = true;
+            this.trigger('userActive');
+        }
+    }
+
+    userInactive(): void {
+        clearTimeout(this.activeTimeout);
+        this.activeTimeout = -1;
+        if (this.settingsMenu.visible) {
+            return;
+        }
+        this.inactiveTime = 0;
+        this.showing = false;
+        addClass(this.playerContainer, 'jw-flag-user-inactive');
+        this.trigger('userInactive');
+    }
+
+    addControls(): void {
+        // Put the controls element inside the wrapper
+        if (this.div && this.wrapperElement) {
+            this.wrapperElement.appendChild(this.div);
+        }
+    }
+
+    addBackdrop(): void {
+        // Put the backdrop element on top of overlays during instream mode
+        // otherwise keep it behind captions and on top of preview poster
+        if (this.backdrop && this.wrapperElement) {
+            const element = this.instreamState ? this.div : this.wrapperElement.querySelector('.jw-captions');
+            this.wrapperElement.insertBefore(this.backdrop, element);
+        }
+    }
+
+    removeBackdrop(): void {
+        if (this.backdrop) {
+            const parent = this.backdrop.parentNode;
+            if (parent) {
+                parent.removeChild(this.backdrop);
+            }
+        }
+    }
+
+    controlbarHeight(): number {
+        if (!this.dimensions.cbHeight) {
+            this.dimensions.cbHeight = this.controlbar.element().clientHeight as number;
+        }
+        return this.dimensions.cbHeight;
+    }
+
+    element(): HTMLElement | null {
+        return this.div;
+    }
+
+    resize(): void {
+        this.dimensions = {};
+    }
+
+    setupInstream(): void {
+        this.instreamState = true;
+        // Call Controls.userActivity to display the UI temporarily for the start of the ad
+        this.userActive();
+        this.addBackdrop();
+        if (this.settingsMenu) {
+            this.settingsMenu.close();
+        }
+        removeClass(this.playerContainer, 'jw-flag-autostart');
+        this.controlbar.elements.time.element().setAttribute('tabindex', '-1');
+    }
+
+    destroyInstream(model: ViewModel): void {
+        this.instreamState = null;
+        this.addBackdrop();
+        if (model.get('autostartMuted')) {
+            addClass(this.playerContainer, 'jw-flag-autostart');
+        }
+        this.controlbar.elements.time.element().setAttribute('tabindex', '0');
+    }
+}
+
+interface Controls extends BaseControls {
```

Comment:
same q re: interface with the class present, i believe we can rely on the class as the source of truth here